apps/web/app/(app)/onboarding/actions.ts

```diff
@@ -16,12 +16,19 @@ export async function updateProfileAction(updatedProfile: Partial<TProfileUpdate
   return await updateProfile(session.user.id, updatedProfile);
 }
 
-export async function updateProductAction(productId: string, updatedProduct: Partial<TProductUpdateInput>) {
+export async function updateProductAction(
+  productId: string,
+  updatedProduct: Partial<TProductUpdateInput>,
+  environmentId: string
+) {
   const session = await getServerSession(authOptions);
   if (!session) throw new AuthorizationError("Not authorized");
 
   const isAuthorized = await canUserAccessProduct(session.user.id, productId);
   if (!isAuthorized) throw new AuthorizationError("Not authorized");
 
+  const { hasCreateOrUpdateAccess } = await verifyUserRoleAccess(environmentId, session.user.id);
```

Comment:
we have the same action problem as mentioned below. Please derive environmentId instead of param.